+neuron/PlotShape.m

```diff
@@ -11,67 +11,31 @@
-                % Iterate over segments.
-                segments = s.segments();
-                for j=1:s.nseg
-                    % Get start, pt3ds and end point for each segment.
-                    seg = segments{j};
-                    [x_lo, x_hi] = seg.get_bounds();
-                    x_lo = x_lo * dx;
-                    x_hi = x_hi * dx;
-                    seg_arc_arr = arc3d(arc3d(:) > x_lo & arc3d(:) < x_hi);
-                    seg_arc_arr = [x_lo seg_arc_arr x_hi];
-
-                    % Interpolate x,y,z,d for each point in seg_arc_arr.
-                    seg_x3d_arr = interp1_arr(pt3d(4, :), pt3d(1, :), seg_arc_arr);
-                    seg_y3d_arr = interp1_arr(pt3d(4, :), pt3d(2, :), seg_arc_arr);
-                    seg_z3d_arr = interp1_arr(pt3d(4, :), pt3d(3, :), seg_arc_arr);
-                    seg_d3d_arr = interp1_arr(pt3d(4, :), pt3d(5, :), seg_arc_arr);
-
-                    % Find value of selected quantity for this segment.
-                    var = spi.varname();
-                    seg_value = s.ref(var, seg.x).get();
-                    seg_value_rel = seg_value - spi.low();
-                    seg_value_rel = seg_value_rel  / (spi.high() - spi.low());
-                    seg_value_rel = min(max(seg_value_rel, 0), 1);
-                    
-                    % Plot between pt3ds, one pair at a time.
-                    for k=1:numel(seg_arc_arr)-1
-                        x = [seg_x3d_arr(k) seg_x3d_arr(k+1)];
-                        y = [seg_y3d_arr(k) seg_y3d_arr(k+1)];
-                        z = [seg_z3d_arr(k) seg_z3d_arr(k+1)];
-                        plot_struct = struct;
-                        plot_struct.x = x;
-                        plot_struct.y = y;
-                        plot_struct.z = z;
-                        plot_struct.line_width = (seg_d3d_arr(k) + seg_d3d_arr(k+1))/2;
-                        plot_struct.color = [seg_value_rel, 0, 1-seg_value_rel];
-                        data{end+1} = plot_struct;
-                    end
-                end
-            end
 
+            section_plot_data = double(clib.neuron.get_plot_data(spi));
+            section_plot_data = transpose(reshape(section_plot_data, 9, []));
+
+            x = [section_plot_data(:, 1) section_plot_data(:, 2)];
+            y = [section_plot_data(:, 3) section_plot_data(:, 4)];
+            z = [section_plot_data(:, 5) section_plot_data(:, 6)];
+            d = (section_plot_data(:, 7) + section_plot_data(:, 8)) / 2;
+            v = section_plot_data(:, 9);
+            data = table(x, y, z, d, v, 'VariableNames', {'x', 'y', 'z', 'line_width', 'color'}); 
+
+            % Normalize color value
+            data.color = data.color - spi.low();
+            data.color = data.color / (spi.high() - spi.low());
+            data.color = min(max(data.color, 0), 1);
+
+            % Turn color value into rgb array
+            data.color = [data.color, zeros(size(data.color)), 1-data.color];
```

Comment:
Can we just pass the color value to `plot3` as a number between 0 and 1, and have the user set a color map?

Reply:
No idea if that's technically possible on the MATLAB side, but if someone could dynamically set a color map, I agree: that would be nice.

Reply:
Yes, setting a colormap can be done with: [colormap(target,map)](https://nl.mathworks.com/help/matlab/ref/colormap.html#d126e226936), where we probably want to use the figure as target (instead of an axes). The map can be a predefined [matlab colormap](https://nl.mathworks.com/help/matlab/ref/colormap.html#buc3wsn-1-map) or you can create a custom one.